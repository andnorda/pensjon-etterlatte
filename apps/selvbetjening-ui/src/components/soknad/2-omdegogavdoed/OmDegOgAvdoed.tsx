import SoknadSteg from "../../../typer/SoknadSteg";
import { useTranslation } from "react-i18next";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { ISoekerOgAvdoed } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { FormProvider, useForm } from "react-hook-form";
import { RHFInput } from "../../felles/RHFInput";
import { SkjemaGruppe } from "nav-frontend-skjema";
import ForholdTilAvdoedeSkjema from "./forholdTilAvdoede/ForholdTilAvdoedeSkjema";
import Feilmeldinger from "../../felles/Feilmeldinger";
import Datovelger from "../../felles/Datovelger";
import { Cell, Grid, Label, Title } from "@navikt/ds-react";
import { RHFSpoersmaalRadio } from "../../felles/RHFRadio";
import NySivilstatus from "./nySivilstatus/NySivilstatus";
import Navigasjon from "../../felles/Navigasjon";
import HvorforSpoerVi from "../../felles/HvorforSpoerVi";
import { isEmpty } from "lodash";
import useEffectOnce from "../../../hooks/useEffectOnce";

const OmDegOgAvdoed: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const lagre = (data: ISoekerOgAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: data })
        neste!!()
    };

    const methods = useForm<ISoekerOgAvdoed>({
        defaultValues: state.omDegOgAvdoed || {},
        shouldUnregister: true
    });

    useEffectOnce(() => {
        methods.reset(state.omDegOgAvdoed);
    }, !isEmpty(state.omDegOgAvdoed));

    const {
        handleSubmit,
        formState: { errors }
    } = methods;

    return (
        <>
            {/* Steg 2 */}
            <SkjemaGruppe>
                <Title size={"m"} className={"center"}>
                    {t("omDegOgAvdoed.tittel")}
                </Title>
            </SkjemaGruppe>

            {/* Skjema for utfylling av info om innlogget bruker / s√∏ker */}
            <FormProvider {...methods}>
                <form>
                    <SkjemaGruppe>
                        <Label>{t("omDegOgAvdoed.avdoed.hvem")}</Label>

                        <Grid>
                            <Cell xs={12} md={6}>
                                <RHFInput
                                    name={"avdoed.fornavn"}
                                    label={t("omDegOgAvdoed.avdoed.fornavn")}
                                />
                            </Cell>

                            <Cell xs={12} md={6}>
                                <RHFInput
                                    name={"avdoed.etternavn"}
                                    label={t("omDegOgAvdoed.avdoed.etternavn")}
                                />
                            </Cell>
                        </Grid>
                    </SkjemaGruppe>

                    <SkjemaGruppe>
                        <Label>{t("omDegOgAvdoed.avdoed.datoForDoedsfallet")}</Label>

                        <Datovelger
                            name={"avdoed.datoForDoedsfallet"}
                            label={t("omDegOgAvdoed.avdoed.dato")}
                            maxDate={new Date()}
                        />
                    </SkjemaGruppe>

                    <RHFSpoersmaalRadio
                        name={"avdoed.doedsfallAarsak"}
                        legend={t("omDegOgAvdoed.avdoed.doedsfallAarsak")}
                        description={
                            <HvorforSpoerVi>{t("omDegOgAvdoed.avdoed.doedsfallAarsakHvorfor")}</HvorforSpoerVi>}
                        vetIkke
                    />

                    {/* 2.9 */}
                    <ForholdTilAvdoedeSkjema/>

                    <NySivilstatus/>

                    <Feilmeldinger errors={errors}/>

                    <Navigasjon
                        forrige={{ onClick: forrige }}
                        neste={{ onClick: handleSubmit(lagre) }}
                    />
                </form>
            </FormProvider>
        </>
    );
};

export default OmDegOgAvdoed;
