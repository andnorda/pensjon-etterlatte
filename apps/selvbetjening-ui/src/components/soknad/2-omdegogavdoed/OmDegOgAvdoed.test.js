import { render } from "@testing-library/react";
import uuid from "uuid";
import lodash from "lodash";
import * as JSutils from "nav-frontend-js-utils";
import OmDegOgAvdoed from "./OmDegOgAvdoed";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

JSutils.guid = jest.fn(() => "123");
describe("Om deg og avdÃ¸d", () => {
    it("Snapshot", () => {
        const { container } = render(<OmDegOgAvdoed />);
        expect(container).toMatchSnapshot();
    });
});
