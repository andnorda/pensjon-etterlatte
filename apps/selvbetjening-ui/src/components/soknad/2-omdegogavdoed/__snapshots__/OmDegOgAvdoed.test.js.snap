// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om deg og avdød Snapshot med sivilstatus samboer 1`] = `
<div>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDegOgAvdoed.tittel
    </h1>
    <div
      aria-live="polite"
    />
  </fieldset>
  <form>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.hvem
      </p>
      <div
        class="navds-grid"
      >
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.fornavn"
                placeholder="omDegOgAvdoed.avdoed.fornavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.etternavn"
                placeholder="omDegOgAvdoed.avdoed.etternavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.datoForDoedsfallet
      </p>
      <section
        class="skjemaelement undefined"
      >
        <label
          class="skjemaelement__label"
          for="avdoed.datoForDoedsfallet"
        >
          omDegOgAvdoed.avdoed.dato
        </label>
        <div
          class="datovelger"
        >
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                autocomplete="off"
                class="skjemaelement__input dato-input"
                id="avdoed.datoForDoedsfallet"
                placeholder="dd.mm.åååå"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="kalender-ikon"
            tabindex="0"
          >
            <svg
              fill="none"
              height="24px"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6 7V5H2v5h20V5h-4v2a1 1 0 11-2 0V5H8v2a1 1 0 11-2 0zm10-4H8V1a1 1 0 10-2 0v2H2a2 2 0 00-2 2v17a2 2 0 002 2h20a2 2 0 002-2V5a2 2 0 00-2-2h-4V1a1 1 0 10-2 0v2zM2 12v10h20V12H2zm6 3a1 1 0 00-1-1H5a1 1 0 100 2h2a1 1 0 001-1zm-1 3a1 1 0 110 2H5a1 1 0 110-2h2zm6-4h-2a1 1 0 100 2h2a1 1 0 100-2zm-2 4h2a1 1 0 110 2h-2a1 1 0 110-2zm9-3a1 1 0 00-1-1h-2a1 1 0 100 2h2a1 1 0 001-1zm-4 4a1 1 0 011-1h2a1 1 0 110 2h-2a1 1 0 01-1-1z"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        
      </section>
      <div
        aria-live="polite"
      />
    </fieldset>
    <div
      id="avdoed.doedsfallAarsak"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe inputPanelGruppe inline"
      >
        <legend
          class="skjemagruppe__legend"
        >
          omDegOgAvdoed.avdoed.doedsfallAarsak
        </legend>
        <div
          class="skjemagruppe__description"
        >
          <div
            class="hvorforPanel"
          >
            <button
              aria-expanded="false"
              class="hvorforPanel__toggle"
              data-testid="hvorforPanel_toggle"
              type="button"
            >
              <span>
                hvorforSpoerVi
              </span>
              <span>
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 17L3 8.429 4.5 7l7.5 7.143L19.5 7 21 8.429 12 17z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="inputPanelGruppe__inner"
        >
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="avdoed.doedsfallAarsak"
              required=""
              type="radio"
              value="Ja"
            />
            <span
              class="inputPanel__label"
            >
              radiobuttons.ja
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="avdoed.doedsfallAarsak"
              required=""
              type="radio"
              value="Nei"
            />
            <span
              class="inputPanel__label"
            >
              radiobuttons.nei
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="avdoed.doedsfallAarsak"
              required=""
              type="radio"
              value="Vet ikke"
            />
            <span
              class="inputPanel__label"
            >
              radiobuttons.vetIkke
            </span>
          </label>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
    </div>
    <div
      id="forholdTilAvdoede.relasjon"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe inputPanelGruppe"
      >
        <legend
          class="skjemagruppe__legend"
        >
          omDegOgAvdoed.forholdTilAvdoede.relasjon
        </legend>
        <div
          class="inputPanelGruppe__inner"
        >
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.gift"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.gift
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.separert"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.separert
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.samboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.samboer
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.skilt"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.skilt
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.tidligereSamboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.tidligereSamboer
            </span>
          </label>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
    </div>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDegOgAvdoed.nySivilstatus.sivilstatus
        </h1>
        <p
          class="navds-body-long"
        >
          omDegOgAvdoed.nySivilstatus.beskrivelse
        </p>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        id="nySivilstatus.sivilstatus"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe inputPanelGruppe"
        >
          <div
            class="inputPanelGruppe__inner"
          >
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ingen"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ingen
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ekteskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ekteskap
              </span>
            </label>
            <label
              class="inputPanel radioPanel inputPanel--checked"
            >
              <input
                aria-checked="true"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.samboerskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.samboerskap
              </span>
            </label>
          </div>
          <div
            aria-live="polite"
          />
        </fieldset>
      </div>
      <div
        class="navds-panel navds-panel--border"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe"
        >
          <h1
            class="navds-heading navds-heading--small"
          >
            omDegOgAvdoed.nySivilstatus.samboerskap.samboer.tittel
          </h1>
          <div
            aria-live="polite"
          />
        </fieldset>
        <div
          class="navds-grid"
        >
          <div
            class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
          >
            <div
              class="kol-50"
            >
              <div
                class="skjemaelement"
              >
                <input
                  aria-invalid="false"
                  class="skjemaelement__input input--fullbredde"
                  id="nySivilstatus.samboerskap.samboer.fornavn"
                  placeholder="omDegOgAvdoed.nySivilstatus.samboerskap.samboer.fornavn"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
          >
            <div
              class="kol-50"
            >
              <div
                class="skjemaelement"
              >
                <input
                  aria-invalid="false"
                  class="skjemaelement__input input--fullbredde"
                  id="nySivilstatus.samboerskap.samboer.etternavn"
                  placeholder="omDegOgAvdoed.nySivilstatus.samboerskap.samboer.etternavn"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="skjemaelement"
        >
          <label
            class="skjemaelement__label"
            for="nySivilstatus.samboerskap.samboer.foedselsnummer"
          >
            felles.fnr
          </label>
          <input
            aria-invalid="false"
            class="skjemaelement__input input--s"
            id="nySivilstatus.samboerskap.samboer.foedselsnummer"
            placeholder="felles.fnrPlaceholder"
            required=""
            rules="[object Object]"
            type="tel"
            value=""
          />
        </div>
        <div
          id="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
        >
          <fieldset
            aria-invalid="false"
            class="skjemagruppe inputPanelGruppe inline"
          >
            <legend
              class="skjemagruppe__legend"
            >
              omDegOgAvdoed.nySivilstatus.samboerskap.hattBarnEllerVaertGift
            </legend>
            <div
              class="inputPanelGruppe__inner"
            >
              <label
                class="inputPanel radioPanel"
              >
                <input
                  aria-checked="false"
                  aria-invalid="false"
                  class="inputPanel__field"
                  name="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
                  required=""
                  type="radio"
                  value="Ja"
                />
                <span
                  class="inputPanel__label"
                >
                  radiobuttons.ja
                </span>
              </label>
              <label
                class="inputPanel radioPanel"
              >
                <input
                  aria-checked="false"
                  aria-invalid="false"
                  class="inputPanel__field"
                  name="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
                  required=""
                  type="radio"
                  value="Nei"
                />
                <span
                  class="inputPanel__label"
                >
                  radiobuttons.nei
                </span>
              </label>
            </div>
            <div
              aria-live="polite"
            />
          </fieldset>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-footer"
    >
      <div
        class="navigasjon-rad"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.neste
          </span>
        </button>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-rad"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-button__inner navds-body-short"
        >
          knapp.avbryt
        </span>
      </button>
      <div
        aria-live="polite"
      />
    </fieldset>
  </form>
</div>
`;

exports[`Om deg og avdød Snapshot uten samboerskjema 1`] = `
<div>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDegOgAvdoed.tittel
    </h1>
    <div
      aria-live="polite"
    />
  </fieldset>
  <form>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.hvem
      </p>
      <div
        class="navds-grid"
      >
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.fornavn"
                placeholder="omDegOgAvdoed.avdoed.fornavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="skjemaelement"
            >
              <input
                aria-invalid="false"
                class="skjemaelement__input input--fullbredde"
                id="avdoed.etternavn"
                placeholder="omDegOgAvdoed.avdoed.etternavn"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <p
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.datoForDoedsfallet
      </p>
      <section
        class="skjemaelement undefined"
      >
        <label
          class="skjemaelement__label"
          for="avdoed.datoForDoedsfallet"
        >
          omDegOgAvdoed.avdoed.dato
        </label>
        <div
          class="datovelger"
        >
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                autocomplete="off"
                class="skjemaelement__input dato-input"
                id="avdoed.datoForDoedsfallet"
                placeholder="dd.mm.åååå"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="kalender-ikon"
            tabindex="0"
          >
            <svg
              fill="none"
              height="24px"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6 7V5H2v5h20V5h-4v2a1 1 0 11-2 0V5H8v2a1 1 0 11-2 0zm10-4H8V1a1 1 0 10-2 0v2H2a2 2 0 00-2 2v17a2 2 0 002 2h20a2 2 0 002-2V5a2 2 0 00-2-2h-4V1a1 1 0 10-2 0v2zM2 12v10h20V12H2zm6 3a1 1 0 00-1-1H5a1 1 0 100 2h2a1 1 0 001-1zm-1 3a1 1 0 110 2H5a1 1 0 110-2h2zm6-4h-2a1 1 0 100 2h2a1 1 0 100-2zm-2 4h2a1 1 0 110 2h-2a1 1 0 110-2zm9-3a1 1 0 00-1-1h-2a1 1 0 100 2h2a1 1 0 001-1zm-4 4a1 1 0 011-1h2a1 1 0 110 2h-2a1 1 0 01-1-1z"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        
      </section>
      <div
        aria-live="polite"
      />
    </fieldset>
    <div
      id="avdoed.doedsfallAarsak"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe inputPanelGruppe inline"
      >
        <legend
          class="skjemagruppe__legend"
        >
          omDegOgAvdoed.avdoed.doedsfallAarsak
        </legend>
        <div
          class="skjemagruppe__description"
        >
          <div
            class="hvorforPanel"
          >
            <button
              aria-expanded="false"
              class="hvorforPanel__toggle"
              data-testid="hvorforPanel_toggle"
              type="button"
            >
              <span>
                hvorforSpoerVi
              </span>
              <span>
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 17L3 8.429 4.5 7l7.5 7.143L19.5 7 21 8.429 12 17z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="inputPanelGruppe__inner"
        >
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="avdoed.doedsfallAarsak"
              required=""
              type="radio"
              value="Ja"
            />
            <span
              class="inputPanel__label"
            >
              radiobuttons.ja
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="avdoed.doedsfallAarsak"
              required=""
              type="radio"
              value="Nei"
            />
            <span
              class="inputPanel__label"
            >
              radiobuttons.nei
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="avdoed.doedsfallAarsak"
              required=""
              type="radio"
              value="Vet ikke"
            />
            <span
              class="inputPanel__label"
            >
              radiobuttons.vetIkke
            </span>
          </label>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
    </div>
    <div
      id="forholdTilAvdoede.relasjon"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe inputPanelGruppe"
      >
        <legend
          class="skjemagruppe__legend"
        >
          omDegOgAvdoed.forholdTilAvdoede.relasjon
        </legend>
        <div
          class="inputPanelGruppe__inner"
        >
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.gift"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.gift
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.separert"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.separert
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.samboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.samboer
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.skilt"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.skilt
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="forholdTilAvdoede.relasjon"
              required=""
              type="radio"
              value="avdoede.relasjon.tidligereSamboer"
            />
            <span
              class="inputPanel__label"
            >
              avdoede.relasjon.tidligereSamboer
            </span>
          </label>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
    </div>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe skjemagruppering"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDegOgAvdoed.nySivilstatus.sivilstatus
        </h1>
        <p
          class="navds-body-long"
        >
          omDegOgAvdoed.nySivilstatus.beskrivelse
        </p>
        <div
          aria-live="polite"
        />
      </fieldset>
      <div
        id="nySivilstatus.sivilstatus"
      >
        <fieldset
          aria-invalid="false"
          class="skjemagruppe inputPanelGruppe"
        >
          <div
            class="inputPanelGruppe__inner"
          >
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ingen"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ingen
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.ekteskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.ekteskap
              </span>
            </label>
            <label
              class="inputPanel radioPanel"
            >
              <input
                aria-checked="false"
                aria-invalid="false"
                class="inputPanel__field"
                name="nySivilstatus.sivilstatus"
                required=""
                type="radio"
                value="nySivilstatus.samboerskap"
              />
              <span
                class="inputPanel__label"
              >
                nySivilstatus.samboerskap
              </span>
            </label>
          </div>
          <div
            aria-live="polite"
          />
        </fieldset>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-footer"
    >
      <div
        class="navigasjon-rad"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-button__inner navds-body-short"
          >
            knapp.neste
          </span>
        </button>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-rad"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-button__inner navds-body-short"
        >
          knapp.avbryt
        </span>
      </button>
      <div
        aria-live="polite"
      />
    </fieldset>
  </form>
</div>
`;
